batch/batch/driver/scheduler.py

```diff
@@ -188,9 +201,14 @@ def allocate_cores(user, mark):
 
                 if record['cancel']:
                     log.info(f'cancelling job {id}')
-                    await mark_job_complete(self.app, batch_id, job_id, None, None,
-                                            'Cancelled', None, None, None, 'cancelled')
                     should_wait = False
+
+                    async_work.append(
+                        LogAndIgnoreErrors(
```

Comment:
This seems overly complex to me, too.  I'd just make a local async function with the exception handling (duplicated).  You're just saving 6 lines of try/except/log.  If you want a helper, what about:

```
async def with_error_handling(description, f, *args, **kwargs):
  try:
    return await f(*args, **kwargs);
  except blah blah log
```

and

```
  async_work.append(functools.partial(with_error_handling, 'description', mark_job_complete, args...))
```